### ISSUE TITLE
Ability to set a maintenance page for instance downtime

### ISSUE THREAD

**Issue by davidwolfpaw** [tootsuite/mastodon#15556]
### Pitch

The ability to set a maintenance page to display when there is downtime on an instance (in which the server is still running but the instance itself is not) would be a useful addition. This would allow instance owners to inform users that the instance still exists, but that there is some reason for it to be down. This would be more useful than the default error page that gives no indication of what is wrong.

### Motivation

There are a few times where an instance needs to be taken down temporarily. While @Gargron has previously pointed out that most updates are downtime-free, this does not address the times where that is not true. Likewise it does not address times where there would be planned downtime, such as running specific backup, resize, or maintenance tasks.

**Comment by davidstown** [comment]
I agree. Kinda late but this has not still been added and it needs to be added. 
Also add the ability to set that's in maintenance manually.
